### PR TITLE
FE: Landing Page UI

### PR DESCRIPTION
**Pull request by Habib-Ahmad**
### What this PR does (required):
- A landing page component was created and routed to the base url
- The Navbar was made responsive

### Screenshots / Videos (front-end only):
image

### Any information needed to test this feature (required):
- Go to the base url

### Any issues with the current functionality (optional):
- The dashboard cannot be accessed while logged out

### CODE REVIEW

File: client/src/pages/LandingPage/LandingPage.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+          <Button type="submit" variant="contained" size="large" className={classes.submitBtn}>
+            Find my dog sitter
+          </Button>
+        </Box>
+      </Grid>
+
+      <Grid item xs={12} md={6}>
+        <Box className={classes.imageWrapper}>
+          <img src={landingPage} alt="" className={classes.image} />
```
Comment (sundayezeilo) [inline client/src/pages/LandingPage/LandingPage.tsx]: It's a good practice to add the `alt` attribute to `img` element.

**Review by sundayezeilo — COMMENTED** [review]
Good work with this one :+1:

**Comment by moffatethan** [comment]
Just a comment on your PR description. Feel free to be more descriptive (ie, talk about new components created). Linking the issue is a good one as well.

**Comment by moffatethan** [comment]
Let's work on the mobile design of this 😃. Now that the page is white - any chance we could touch it up. Thanks!

Screen Shot 2022-01-14 at 5 10 34 PM

File: client/src/components/FormInput/FormInput.tsx [hunk 2]
```diff
@@ -23,6 +23,8 @@ interface FormInputProps {
   selectOptions?: Array<{ label: string; value: string }> | undefined;
   selectValue?: string;
 
+  placeholder?: string;
+
```
Comment (moffatethan) [inline client/src/components/FormInput/FormInput.tsx]: Underneath we are already passing `inputProps` to the interface. This is essentially all inputProps. You can actually remove this and just pass in the `placeholder` property to the `FormInput` and it will work the same way.

File: client/src/components/FormInput/FormInput.tsx [hunk 3]
```diff
@@ -58,7 +61,7 @@ const FormInput: React.FC<FormInputProps> = ({
           ))}
         </Select>
       ) : (
-        <StyledInput defaultValue={defaultValue} id={id} {...rest} />
+        <StyledInput placeholder={placeholder} defaultValue={defaultValue} id={id} {...rest} />
```
Comment (moffatethan) [inline client/src/components/FormInput/FormInput.tsx]: This `placeholder` property can be removed if we remove the placeholder prop from the parent.

**Comment by moffatethan** [comment]
Also, looks like navbar changes are causing bad alignment in the other pages. We should fix this as well please.

Screen Shot 2022-01-14 at 5 19 26 PM

File: client/src/components/Navbar/Navbar.tsx [hunk 4]
```diff
@@ -84,7 +84,10 @@ const MenuItem: React.FC<{
 
   return (
     <Grid key={resource} sx={{ textAlign: 'center' }} xs={2} justifySelf="flex-end" item>
-      <NavLink className={classes.navbarItem} to={resource}>
+      <NavLink
+        className={clsx(classes.navbarItem, location.pathname === '/' && classes.transparentNavbarItem)}
+        to={resource}
+      >
```
Comment (moffatethan) [inline client/src/components/Navbar/Navbar.tsx:93]: 💯

File: client/src/context/useAuthContext.tsx [hunk 5]
```diff
@@ -58,7 +58,7 @@ export const AuthProvider: FunctionComponent = ({ children }): JSX.Element => {
           // don't need to provide error feedback as this just means user doesn't have saved cookies or the cookies have not been authenticated on the backend
           setLoggedInUser(null);
           if (!(history.location.pathname === '/signup')) {
-            history.push('/login');
+            history.push('/');
```
Comment (moffatethan) [inline client/src/context/useAuthContext.tsx]: How come we are redirecting back to the landing page rather than the login page if the user isn't signed in? Let's revert this change 😃

File: client/src/pages/LandingPage/useStyles.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import { Theme } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+
+export const useStyles = makeStyles((theme: Theme) => ({
+  left: {
+    padding: theme.spacing(15, 15, 0),
```
Comment (moffatethan) [inline client/src/pages/LandingPage/useStyles.ts]: Let's work on the padding based on the breakpoints, smaller screen this is `120px`

**Review by moffatethan — CHANGES_REQUESTED** [review]
Good work on this, code looks good. Just want you to spend some time on the UI on smaller screens. Be careful when editing sizes on components that are mutual across other pages as when I signed into my demo account I saw the navbar alignment looked off center.

**Review by moffatethan — APPROVED** [review]
Good work, thanks for doing those changes!
